### PR TITLE
MYEM-3835 added function to get single template and be able to create new one based on it

### PR DESCRIPTION
**Pull request by youcefTab**

### CODE REVIEW

File: notification_lib/novu_manager/generic_novu_manager.py [hunk 1]
```diff
@@ -51,3 +51,49 @@ def get_generic_novu_templates(cls) -> list[dict[str, Any]]:
...
+            url=f"{cls.api_url_generic_novu}/v1/notification-templates/{template_id}",
+            headers={"Authorization": f"Bearer {cls.get_novu_jwt_bearer_token()}"},
+            timeout=5,
+        )
+
+        if response.status_code // 100 == 2:
+            result = response.json()["data"]
+            steps = []
+
+            def construct_filters(step_filter: dict[str, Any]) -> dict[str, Any]:
```
Comment (ouss1919) [inline notification_lib/novu_manager/generic_novu_manager.py:68]: why this function is inside get_generic_template_by_id function

**Review by ouss1919 — APPROVED** [review]

File: notification_lib/novu_manager/generic_novu_manager.py (hunk 1 above)
Comment (youcefTab) [reply notification_lib/novu_manager/generic_novu_manager.py:68]: thought about it like a small function for formating, will change it to a static function

File: notification_lib/novu_manager/generic_novu_manager.py [hunk 2]
```diff
@@ -51,3 +51,50 @@ def get_generic_novu_templates(cls) -> list[dict[str, Any]]:
...
+    @staticmethod
+    def format_filter(step_filter: dict[str, Any]) -> dict[str, Any]:
+        """Take useful fields for the filter."""
+
+        needed_fields = ["children", "isNegated", "type", "value"]
+        formated_filter = {}
+        for key in step_filter.keys():
+            if key in needed_fields:
+                formated_filter[key] = step_filter[key]
+        return formated_filter
```
Comment (fchennouf) [inline notification_lib/novu_manager/generic_novu_manager.py:64]: return {key: step_filter[key] for key in step_filter.keys() if key in ["children", "isNegated", "type", "value"]}

File: notification_lib/novu_manager/generic_novu_manager.py [hunk 3]
```diff
@@ -51,3 +51,50 @@ def get_generic_novu_templates(cls) -> list[dict[str, Any]]:
...
+    @classmethod
+    def get_generic_template_by_id(cls, template_id: str) -> dict[str, Any]:
+        """Get generic template by providing his id."""
+        response = requests.get(
+            url=f"{cls.api_url_generic_novu}/v1/notification-templates/{template_id}",
+            headers={"Authorization": f"Bearer {cls.get_novu_jwt_bearer_token()}"},
+            timeout=5,
+        )
+
+        if response.status_code // 100 == 2:
```
Comment (fchennouf) [inline notification_lib/novu_manager/generic_novu_manager.py:75]: use specific code resposnse

**Pull request merged**